src/sqlfluff/core/dialects/dialect_snowflake.py

```diff
@@ -144,29 +140,23 @@
 
 
 @snowflake_dialect.segment(replace=True)
-class StatementSegment(ansi_StatementSegment):
+class StatementSegment(ansi.StatementSegment):
     """A generic segment, to any of its child subsegments."""
 
-    type = "statement"
-    match_grammar = GreedyUntil(Ref("SemicolonSegment"))
-
-    parse_grammar = OneOf(
-        Ref("SelectableGrammar"),
-        Ref("InsertStatementSegment"),
-        Ref("TransactionStatementSegment"),
-        Ref("DropStatementSegment"),
-        Ref("AccessStatementSegment"),
-        Ref("CreateTableStatementSegment"),
-        Ref("AlterTableStatementSegment"),
-        Ref("CreateViewStatementSegment"),
-        Ref("DeleteStatementSegment"),
-        Ref("UpdateStatementSegment"),
-        Ref("CreateModelStatementSegment"),
-        Ref("DropModelStatementSegment"),
-        Ref("UseStatementSegment"),
-        Ref("CreateStatementSegment"),
-        Ref("CreateCloneStatementSegment"),
-        Ref("ShowStatementSegment"),
+    parse_grammar = ansi.StatementSegment.parse_grammar.copy(
+        insert=[
+            Ref("UseStatementSegment"),
+            Ref("CreateStatementSegment"),
+            Ref("CreateCloneStatementSegment"),
+            Ref("ShowStatementSegment"),
+        ],
+        remove=[
```

Comment:
Thinking ahead a bit, places where we use `remove` might suggest things we'd want to remove from the base dialect. (I think several of these removals are Postgres-specific stuff.) Think it's worth creating an issue for this?

Reply:
I'm tempted to make those issues when they come up, but in this case I think it's set up how it should be. It's frustrating that snowflake takes *most* of it's queues from postgres, but does remove some elements too. 🤷

Reply:
I agree, low priority, longer term stuff. 👍